python/tests/test_metadata.py

```diff
@@ -440,15 +440,23 @@ def test_bad_top_level_type(self):
                 metadata.MetadataSchema(schema)
 
     def test_null_union_top_level(self):
-        schema = {
-            "codec": "json",
-            "type": ["object", "null"],
-            "properties": {"one": {"type": "string"}, "two": {"type": "number"}},
-        }
-        ms = metadata.MetadataSchema(schema)
-        row_data = {"one": "tree", "two": 5}
-        assert ms.decode_row(ms.validate_and_encode_row(row_data)) == row_data
-        assert ms.decode_row(ms.validate_and_encode_row(None)) is None
+        for codec in ["json", "struct"]:
```

Comment:
This was a test I added at first before I figured out where the problem was, and figured it's a good one, so I'd leave it in.

Reply:
Good to pytest.mark.parametrize this rather than looping